File: 태성/다트게임2.py
dartResult = "1S2D*3T"

def solution(dartResult):
    answer = 0

    pre = []
    tmp = ""
    for d in dartResult:
        if d not in ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]:
            if tmp:
                pre.append(int(tmp))
                tmp = ""
                pre.append(d)
            else:
                pre.append(d)
        else:
            tmp += d
    print(pre)


    array = []

    for p in pre:
        if p in [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]:
            if len(array) < 2:
                array.append(p)
            else:
                answer += array.pop(0)
                array.append(p)

        else:
            if p == "S":
               continue

            elif p == "D":
                dd = array.pop()
                array.append(dd * dd)

            elif p == "T":
                t = array.pop()
                array.append(t * t * t)


            elif p == "*":
                if len(array) == 1:
                    j = array.pop()
                    array.append(j * 2)
                elif len(array) == 2:
                    j = array.pop()
                    jj = array.pop()

                    array.append(jj * 2)
                    array.append(j * 2)

            elif p == "#":
                z = array.pop()
                array.append(z * (-1))

    for a in array:
        answer += a


    return answer

print(solution(dartResult))

File: 태성/0615/다트게임2.js
const dartResult = "1D2S#10S"

function solution(dartResult) {
  var answer = 0;
  var pre = []
  var tmp = ""
  const chk_list = ["S", "D", "T", "#", "*"]
  for (let d of dartResult) {
    // console.log(d)
    if (chk_list.includes(d)) {
      if (tmp.length > 0) {
        pre.push(parseInt(tmp))
        tmp = ""
        // console.log(pre)
      }
      // console.log(d)
      pre.push(d)
    } else{
      // pre.push(d)
      tmp += d
    }
    // console.log(tmp)
  }

  console.log(pre)

  var array = []
  for (let p of pre) {
    // console.log("p")
    if (chk_list.includes(p)) {
      console.log(p)
      if (p === "S") {
        var s = array.pop()
        array.push(s)
      } else if (p === "D") {
        var dd = array.pop()
        console.log(dd)
        array.push(dd*dd)
        console.log(array)
      } else if (p === "T") {
        var tt = array.pop()
        array.push(tt*tt*tt)
      } else if (p === "*") {
        if (array.length === 1) {
          var star = array.pop()
          array.push(star*2)
        } else if (array.length === 2) {
          var star_2 = array.pop()
          var star_1 = array.pop()

          array.push(star_1*2)
          array.push(star_2*2)
        }
      } else if (p === "#") {
        var mm = array.pop()
        array.push(mm*(-1))
        
      }
    } else {
      console.log("숫자")
      console.log(array)
      if (array.length < 2) {
        array.push(p)
        console.log(array)
      } else if (array.length === 2) {
        answer += array.shift()
        console.log(answer)
        console.log(array)
        array.push(p)
        console.log(array)
      }
    }
  }
  for (let a of array) {
    console.log(answer)
    answer += a
  }
  console.log(answer)
  return answer;
}


solution(dartResult)